server/src/main/java/org/elasticsearch/action/search/SearchResponseSections.java

```diff
@@ -8,30 +8,20 @@
 
 package org.elasticsearch.action.search;
 
-import org.elasticsearch.common.collect.Iterators;
-import org.elasticsearch.common.io.stream.StreamOutput;
-import org.elasticsearch.common.xcontent.ChunkedToXContent;
 import org.elasticsearch.search.SearchHits;
 import org.elasticsearch.search.aggregations.Aggregations;
 import org.elasticsearch.search.profile.SearchProfileResults;
 import org.elasticsearch.search.profile.SearchProfileShardResult;
 import org.elasticsearch.search.suggest.Suggest;
-import org.elasticsearch.xcontent.ToXContent;
 
-import java.io.IOException;
 import java.util.Collections;
-import java.util.Iterator;
 import java.util.Map;
 
 /**
- * Base class that holds the various sections which a search response is
- * composed of (hits, aggs, suggestions etc.) and allows to retrieve them.
- *
- * The reason why this class exists is that the high level REST client uses its own classes
- * to parse aggregations into, which are not serializable. This is the common part that can be
- * shared between core and client.
+ * Holds some sections that a search response is composed of (hits, aggs, suggestions etc.) during some steps of the search response
```

Comment:
Does it make sense to mark this class as deprecated?

Reply:
Maybe not, I was trying to remove it altogether but the few use cases where this is used response merging logic actually look nicer having a record style object like this rather than a bunch of parameters I think. I was gonna double check that point though, maybe not all of these fields are needed and that becomes not true anymore :) lets see 🤞